clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/CreateAzureServerGroupAtomicOperation.groovy

```diff
@@ -96,43 +96,46 @@ class CreateAzureServerGroupAtomicOperation implements AtomicOperation<Map> {
       description.clusterName = description.getClusterName()
       description.appName = description.application
 
-      task.updateStatus(BASE_PHASE, "Deploying new load balancer from ${description.loadBalancerName ?: 'new'}")
-      AzureLoadBalancerDescription lbDescription = description.loadBalancerName ?
-        description.credentials.networkClient.getLoadBalancer(resourceGroupName, description.loadBalancerName) : new AzureLoadBalancerDescription()
-      lbDescription.loadBalancerName = description.name
-      lbDescription.region = description.region
-      lbDescription.accountName = description.accountName
-      lbDescription.credentials = description.credentials
-      lbDescription.appName = description.application
-      lbDescription.cluster = description.clusterName
-      lbDescription.serverGroup = description.name
-      lbDescription.stack = description.stack
-      lbDescription.detail = description.detail
-      lbDescription.securityGroup = description.securityGroup?.name
-      lbDescription.vnet = virtualNetworkName
-      lbDescription.subnet = subnetName
-
-      task.updateStatus(BASE_PHASE, "Create new load balancer ${description.loadBalancerName} in ${description.region}...")
-      DeploymentExtended deployment = description.credentials.resourceManagerClient.createResourceFromTemplate(description.credentials,
-        AzureLoadBalancerResourceTemplate.getTemplate(lbDescription),
-        resourceGroupName,
-        description.region,
-        description.loadBalancerName,
-        "loadBalancer")
-      errList = AzureDeploymentOperation.checkDeploymentOperationStatus(task, BASE_PHASE, description.credentials, resourceGroupName, deployment.name)
+      // get the app gateway. Verify that it can be used for this server group/cluster. add an app pool if it doesn't already exist
+      def appGateway = description.credentials.networkClient.getAppGateway(resourceGroupName, description.loadBalancerName)
+      if (appGateway) {
+        if (appGateway.tags?.cluster && appGateway.tags?.cluster != description.clusterName) {
```

Comment:
"_this is "business logic" outside the scope of the responsibilities of the clients and should remain that way._"
Though I might agree with the principal you're trying to set here, I don't think we could do that anyway; for instance appGateway.tags.cluster will be set via a "client" call anyway. But let's not worry about this for now, I just wanted to give you a heads up that this might get refactored later on.